src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/softmax_gpu_blocked.cl

```diff
@@ -0,0 +1,101 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "include/batch_headers/common.cl"
+
+
+KERNEL(softmax)(
+    __global INPUT0_TYPE* input,
```

Comment:
Do we really need separate kernel for blocked layouts? As I can see, it's not that different comparing to ref version, so maybe this impl can just replace ref one?

Reply:
I thought of this, but decided to make it as a separate kernel. 
First of all, GET_INDEX* macros may decrease performance, so anyway it is a good idea to keep existing softmax_gpu_ref.cl as is - just consder it as a kernel optimized for plain layout.
Another reason is that ref kernel itself does not look so straightforward, so bringing additional #ifdefs to separate plain/blocked branches would make readability even worse.

Reply:
I think for planar formats expanded GET_INDEX macro will produce pretty much similar code as we have in ref impl. Though maybe we need to check the performance to ensure that no regression will happen. 

As for code complexity, in general I agree, but having 2 kernels may lead to other side effects: increase of kernel selection time and plugin binary size. For single kernel the impact will be negligible, but if we use such approach for other primitives, then the impact may be significant.

Reply:
Merged into one kernel.